Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs

```diff
@@ -976,6 +978,10 @@ internal struct LongPathInfo {
                     _analyzer = null;
                 }
 
+                if (_idleNodeModulesTimer != null) {
+                    _idleNodeModulesTimer.Dispose();

```

Comment:
Additionally just looked at the code again, there are a couple other issues we'll need to sort out. 
* Additionally, we should dispose the timer and unsubscribe under a lock to prevent race conditions (you can use the _idleNodeModulesLock)
* also set the timer to null during the dispose process, and check for null before _idleNodeModulesTimer.Change otherwise we'll run into an ObjectDisposedException there.